tools/migration-unified-annotation-versioning/migration.py

```diff
@@ -0,0 +1,529 @@
+            if materialized_version not in layer_version_mapping[tracing_id]:
+                continue
+            new_version = layer_version_mapping[tracing_id][materialized_version]
+            value_bytes = self.get_bytes(collection, tracing_id, materialized_version)
+            if materialized_version != new_version:
+                skeleton = Skeleton.SkeletonTracing()
+                skeleton.ParseFromString(value_bytes)
+                skeleton.version = new_version
+                value_bytes = skeleton.SerializeToString()
+            materialized_versions_unified.append(new_version)
+            self.save_bytes(collection, tracing_id, new_version, value_bytes)
+        return materialized_versions_unified
+
+    def migrate_volume_proto(self, tracing_id: str, layer_version_mapping: LayerVersionMapping, mapping_id_map: MappingIdMap):
+        collection = "volumes"
+        materialized_versions_unified = []
+        materialized_versions = self.list_versions(collection, tracing_id)
+        for materialized_version in materialized_versions:
+            if materialized_version not in layer_version_mapping[tracing_id]:
+                continue
+            new_version = layer_version_mapping[tracing_id][materialized_version]
+            value_bytes = self.get_bytes(collection, tracing_id, materialized_version)
+            if materialized_version != new_version or tracing_id in mapping_id_map:
+                volume = Volume.VolumeTracing()
+                volume.ParseFromString(value_bytes)
+                volume.version = new_version
+                if tracing_id in mapping_id_map:
+                    volume.mappingName = tracing_id
+                value_bytes = volume.SerializeToString()
+            materialized_versions_unified.append(new_version)
+            self.save_bytes(collection, tracing_id, new_version, value_bytes)
+        return materialized_versions_unified
+
+    def list_versions(self, collection, key) -> List[int]:
+        reply = self.src_stub.ListVersions(proto.ListVersionsRequest(collection=collection, key=key))
+        assert_grpc_success(reply)
+        return reply.versions
+
+    def get_bytes(self, collection: str, key: str, version: int) -> bytes:
+        reply = self.src_stub.Get(proto.GetRequest(collection=collection, key=key, version=version))
+        assert_grpc_success(reply)
+        return reply.value
+
+    def save_bytes(self, collection: str, key: str, version: int, value: bytes) -> None:
+        if self.dst_stub is not None:
+            reply = self.dst_stub.Put(proto.PutRequest(collection=collection, key=key, version=version, value=value))
+            assert_grpc_success(reply)
+
+    def migrate_volume_buckets(self, tracing_id: str, layer_version_mapping: LayerVersionMapping):
+        self.migrate_all_versions_and_keys_with_prefix("volumeData", tracing_id, layer_version_mapping, transform_key=self.remove_morton_index)
```

Comment:
What is `self.remove_morton_index` for?

Reply:
Different from the redundant-bboxes issue, I decided that this one can be included here without much additional complexity. https://github.com/scalableminds/webknossos/issues/3546